File: VB/MessageBoxHelper/MyMessageBox/MyXtraMessageArgs.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports System.ComponentModel

Namespace MessageBoxHelper
    Public Class MyXtraMessageArgs
        Inherits XtraMessageBoxArgs

        Public Property ShowCountdown() As Boolean
        Public Property ShowNextTime() As Boolean
        Public Property DisableCancel() As Boolean
        Public Property DisableButtons() As Boolean
        Public Property AutoClose() As Boolean
        Public Property AutoCloseResult() As DialogResult
        Public Property Center() As Boolean
        Public Property Timeout() As Integer
        Public Property ShowMessageNextTime() As Boolean
        Private fButtons As MessageBoxButtons
        Public Shadows Property Buttons() As MessageBoxButtons
            Get
                Return Me.fButtons
            End Get
            Set(ByVal value As MessageBoxButtons)
                MyBase.Buttons = MyXtraMessageBox.GetDialogResultsFromButtons(value)
                Me.fButtons = value
            End Set
        End Property

        Private fIcon As String
        Public Shadows Property Icon() As String
            Get
                Return fIcon
            End Get
            Set(ByVal value As String)
                fIcon = value
                MyBase.Icon = MyXtraMessageBox.GetIcon(value)
            End Set
        End Property

        Public Sub New()
            MyBase.New()
            Me.Buttons = MessageBoxButtons.OK
            Me.ShowCountdown = False
            Me.ShowNextTime = False
            Me.DisableButtons = False
            Me.DisableCancel = False
            Me.AutoClose = False
            Me.Center = False
            Me.Timeout = 0
            Me.ShowMessageNextTime = False
        End Sub
    End Class
End Namespace

File: CS/MessageBoxHelper/MyMessageBox/MyXtraMessageArgs.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using System.ComponentModel;

namespace MessageBoxHelper {
    public class MyXtraMessageArgs : XtraMessageBoxArgs {
        public bool ShowCountdown { get; set; }
        public bool ShowNextTime { get; set; }
        public bool DisableCancel { get; set; }
        public bool DisableButtons { get; set; }
        public bool AutoClose { get; set; }
        public DialogResult AutoCloseResult { get; set; }
        public bool Center { get; set; }
        public int Timeout { get; set; }
        public bool ShowMessageNextTime { get; set; }
        private MessageBoxButtons fButtons;
        public new MessageBoxButtons Buttons {
            get {
                return this.fButtons;
            }
            set {
                base.Buttons = MyXtraMessageBox.GetDialogResultsFromButtons(value);
                this.fButtons = value;
            }
        }

        private string fIcon;
        public new string Icon {
            get {
                return fIcon;
            }
            set {
                fIcon = value;
                base.Icon = MyXtraMessageBox.GetIcon(value);
            }
        }

        public MyXtraMessageArgs()
            : base() {
            this.Buttons = MessageBoxButtons.OK;
            this.ShowCountdown = false;
            this.ShowNextTime = false;
            this.DisableButtons = false;
            this.DisableCancel = false;
            this.AutoClose = false;
            this.Center = false;
            this.Timeout = 0;
            this.ShowMessageNextTime = false;
        }
    }
}
